### ISSUE TITLE
Focus - Click on select/option html menu not scroll in Web browser dock.

### ISSUE THREAD

**Issue by stebulba** [obsproject/obs-browser#388]

**Comment by RytoEX** [comment]
I can confirm this on Windows 10 with OBS Studio 29.

![image](https://user-images.githubusercontent.com/624931/211920099-cf984144-c247-45fe-b21f-844ff62fce0c.png)
![image](https://user-images.githubusercontent.com/624931/211920129-c0324409-7d5b-44f0-8f2d-f3484f729e6a.png)

**Comment by WizardCM** [comment]
I personally expect this is not something we can fix, as the menus in docks are controlled almost exclusively by CEF.

However, it's entirely possible that Qt is passing the wrong data to CEF, or our CEF implementation is reading outdated positional data, resulting in this bug. Additionally (likely related?) normal browsers will close any open select menus when the window is being repositioned; this does not happen with our implementation as CEF doesn't have a proper "unfocus" function we can call.

Long story short, the first time you open a Select menu after a dock or its parent window's position has changed, the menu will be in the wrong spot. Every subsequent open works fine.

I will do some digging later to determine if it's our code or not.

**Comment by WizardCM** [comment]
I decided to test on Linux today, and at least on GNOME, this issue **does not occur**. I expect it's Windows only, but have not yet verified it does not occur on macOS.

**Comment by WizardCM** [comment]
I did some additional debugging today, and have learned nothing useful. Both the coordinates of the CefHost window and the `widget` in `QCefBrowserClient` correctly update their coordinates immediately, as expected. I have no idea why those coordinates aren't being passed to the Select menu.

```cpp
void QCefBrowserClient::OnGotFocus(CefRefPtr<CefBrowser> browser)
{
	QPoint myPos = widget->mapToGlobal(widget->pos());

	RECT rect;
	GetWindowRect(browser->GetHost()->GetWindowHandle(), &rect);
	
	blog(LOG_WARNING, "Refocused, %ix%i // %ix%i", myPos.x(), myPos.y(),
	     rect.left, rect.top);

}
```
